java-checks/src/main/java/org/sonar/java/checks/AnnotationDefaultArgumentCheck.java

```diff
@@ -0,0 +1,87 @@
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.java.checks;
+
+import com.google.common.collect.ImmutableList;
+import org.sonar.check.Rule;
+import org.sonar.java.model.LiteralUtils;
+import org.sonar.java.resolve.JavaSymbol;
+import org.sonar.plugins.java.api.IssuableSubscriptionVisitor;
+import org.sonar.plugins.java.api.semantic.Symbol;
+import org.sonar.plugins.java.api.semantic.Symbol.TypeSymbol;
+import org.sonar.plugins.java.api.tree.AnnotationTree;
+import org.sonar.plugins.java.api.tree.AssignmentExpressionTree;
+import org.sonar.plugins.java.api.tree.ExpressionTree;
+import org.sonar.plugins.java.api.tree.IdentifierTree;
+import org.sonar.plugins.java.api.tree.LiteralTree;
+import org.sonar.plugins.java.api.tree.Tree;
+
+import javax.annotation.Nullable;
+
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+@Rule(key = "S3254")
+public class AnnotationDefaultArgumentCheck extends IssuableSubscriptionVisitor {
+  @Override
+  public List<Tree.Kind> nodesToVisit() {
+    return ImmutableList.of(Tree.Kind.ANNOTATION);
+  }
+
+  @Override
+  public void visitNode(Tree tree) {
+    AnnotationTree annotationTree = (AnnotationTree) tree;
+    TypeSymbol annotationSymbol = annotationTree.symbolType().symbol();
+    if (annotationSymbol.isUnknown()) {
+      return;
+    }
+
+    Map<String, Object> defaultValueByName = annotationSymbol.memberSymbols().stream()
+      .filter(Symbol::isMethodSymbol)
+      .map(s -> (JavaSymbol.MethodJavaSymbol) s)
+      .filter(s -> s.defaultValue() != null)
+      .collect(Collectors.toMap(Symbol::name, JavaSymbol.MethodJavaSymbol::defaultValue));
+
+    for (ExpressionTree argument : annotationTree.arguments()) {
+      Tree valueSet = null;
+      Object defaultValue = null;
+      if (argument.is(Tree.Kind.ASSIGNMENT)) {
+        AssignmentExpressionTree assignmentTree = (AssignmentExpressionTree) argument;
+        IdentifierTree nameTree = (IdentifierTree) assignmentTree.variable();
+        defaultValue = defaultValueByName.get(nameTree.name());
+        valueSet = assignmentTree.expression();
+        if (setValueIsSameAsDefaultValue(defaultValue, valueSet)) {
+          reportIssue(argument, String.format("Remove this default value assigned to parameter \"%s\".", nameTree.name()));
+        }
+      }
+    }
+  }
+
+  private static boolean setValueIsSameAsDefaultValue(@Nullable Object defaultValue, Tree valueSet) {
+    if (valueSet.is(Tree.Kind.STRING_LITERAL)) {
+      return LiteralUtils.trimQuotes(((LiteralTree) valueSet).value()).equals(defaultValue);
+    } else if (valueSet.is(Tree.Kind.INT_LITERAL)) {
+      return LiteralUtils.intLiteralValue((LiteralTree) valueSet).equals(defaultValue);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') A "NullPointerException" could be thrown; "intLiteralValue" is nullable here. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2259)